R/utilities.R

```diff
@@ -101,3 +101,24 @@ loadPages <- function (object, url) {
   }
   return(result)
 }
+
+# We only support a specific format returned by KissMetrics, so we'll use
+# lubridate::fast_strptime instead of parse_date_time.
+# Take a character vector, and if we think it is a time that can be parsed by
+# format we'll return a vector of times, if it can't be converted (any of the
+# non-NA elements fail to parse) we return the original vector
+try_convert_time <- function(char_vector, formats = "%Y-%m-%d %H:%M:%S") {
```

Comment:
Yeah, KM stores things internally as UNIX timestamps. Even when we do the full data export via the S3 bucket we'll still only have to the second fidelity :(

Interestingly enough when we do that export kissmetrics will also write events in the relevant json files based of the order they were received, not the timestamp.

So if the method we use for sending the events to KM is the same we can know the order the events occurred, however for any events that are sent asynchronously the best we can do is the timestamp.